### DISCUSSION TITLE
Codegen w/ regex pattern

### DISCUSSION THREAD

**Discussion by rolanday** [sinclairzx81/typebox#673] [Q&A] [answered]
Hello,

Wanting to use codegen to create string type confirming to regex pattern, like so:

```
/**
 * @pattern ^[a-z0-9]{16}$
 */
export type Cuid = string;
```

Output results in:

```
export type Cuid = Static<typeof Cuid>
export const Cuid = Type.String({ pattern: '^[a' }) // < generated pattern cut off after first hyphen
```

No luck escaping. Any ideas?

> Note: Can copy and paste first code block into https://sinclairzx81.github.io/typebox-workbench/ to repro

Thanks!

**Comment by sinclairzx81** [comment, marked as answer]
@rolanday Hi,

This should be fixed on both the workbench and the `typebox-codegen` project. Note: You will need to quote the pattern as the `-` character terminates the string (where the jsdoc parser assumes a comment will follow)

[Workbench Link Here](https://sinclairzx81.github.io/typebox-workbench/?share=LyoqDQogKiBAcGF0dGVybiAnXlthLXowLTldezE2fSQnDQogKi8NCmV4cG9ydCB0eXBlIEN1aWQgPSBzdHJpbmc7)

```typescript
/**
 * @pattern '^[a-z0-9]{16}$'
 */
export type Cuid = string;
```
Cheers!
S

**Comment by rolanday** [comment]
Thank you so much!! :-) Works like a charm now.

**Comment by rolanday** [comment]
🙏
